tests/zfs-tests/tests/functional/events/zed_fd_spill.ksh

```diff
@@ -0,0 +1,98 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+# DESCRIPTION:
+# Verify ZEDLETs only inherit the fds specified in the manpage
+#
+# STRATEGY:
+# 1. Inject a ZEDLET that dumps the fds it gets to a file.
+# 2. Generate some events.
+# 3. Read back the generated files and assert that there is no fd past 3,
+#    and there are exactly 4 fds.
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/events/events_common.kshlib
+
+verify_runnable "both"
+
+function cleanup
+{
+	log_must rm -rf "$logdir"
+	log_must zed_stop
+}
+
+log_assert "Verify ZEDLETs inherit only the fds specified"
+log_onexit cleanup
+
+logdir="$(mktemp -d)"
+log_must cc -xc -o${ZEDLET_DIR}/all-dumpfds << EOF
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+	if(fork()) {
+		int err;
+		wait(&err);
+		return err;
+	}
+
+	char buf[4096];
+	snprintf(buf, sizeof (buf), "$logdir/%d", getppid());
+	dup2(creat(buf, 0644), STDOUT_FILENO);
+
+	// Fault injexion!
+	snprintf(buf, sizeof (buf), "%d", (getppid() % 20) + 4);
+	puts(buf);
+
+	snprintf(buf, sizeof (buf), "/proc/%d/fd", getppid());
+	execlp("ls", "ls", buf, NULL);
+	_exit(127);
+}
```

Comment:
Is there a reason we couldn't do this with a tiny script for the ZEDLET?  It'd be nice not to have to depend on `cc` here.  The test suite runs with a constrained environment, only the binaries listed in `tests/zfs-tests/include/commands.cfg` are available.

Reply:
I mean, you could, theoretically, do `pid=$$; (ls /proc/$pid/fd)`, right, but it sucks, because you also get all the shell's random fds in there, so you need to filter *those* out somehow, it might mask some stdio fds, &c, &c. Unfortunately, C is the only way to go for this.

You could build this during the normal build process, I guess, but including it in the distribution doesn't sound particularly great. Is there a reason requiring the test system to contain a funxional host C compiler is problematic?

Reply:
> You could build this during the normal build process, I guess, but including it in the distribution doesn't sound particularly great.

All of the tests get included in the `zfs-tests` package, so it's not so bad.  The `tests/zfs-tests/tests/functional/tmpfile/*` tests do something similar and build a couple helper binaries.  Let's go this route.

My aversion to depending on a functional C compiler in the test environment is mainly that this would be the only test which requires it.  It's be nice not to add that dependency.

Reply:
Oh, cool, I didn't realise there were already tests that did this, thanks! Turned out easy enough to adapt that to the zedlet here.